airbyte-integrations/connectors/source-stripe/tools/create_customers.py

```diff
@@ -0,0 +1,212 @@
+        customer_data = customer.dict(exclude_none=True)
+        customer_data.pop("bank_account", {})
+        return HttpRequest(method=HttpMethod.POST, url=url, headers=self._authentication_header, body=customer_data)
+
+    def search_customer(self, customer_name):
+        request = self.prepare_search_customer(customer_name)
+        response = self._requester.submit(request)
+        return response.get("data", [])
+
+    def prepare_create_bank_account(self, customer: Customer):
+        if customer.id is None:
+            raise ValueError(f"Cannot create bank account for uninitialized customer {customer}")
+        if customer.bank_account is None:
+            raise ValueError(f"Cannot create bank account for customer without bank account information: {customer}")
+
+        url = f"https://api.stripe.com/v1/customers/{customer.id}/sources"
+        bank_account_data = {"source": customer.bank_account.dict(exclude_none=True)}
+        bank_account_data["source"]["account_holder_name"] = customer.name
+        return HttpRequest(method=HttpMethod.POST, url=url, headers=self._authentication_header, body=bank_account_data)
+
+    def prepare_search_customer(self, customer_name: str) -> HttpRequest:
+        url = "https://api.stripe.com/v1/customers/search"
+        data = {"query": f"name: '{customer_name}'"}
+        return HttpRequest(method=HttpMethod.GET, url=url, headers=self._authentication_header, body=data)
+
+
+@click.group()
+def _main():
+    pass
+
+
+def kv_translation(d, line, final_str):
+    for key in d:
+        key_str = key if not line else "[{}]".format(key)
+        if type(d[key]) is not dict:
+            final_str = "{}{}{}={}\n".format(final_str, line, key_str, d[key])
+        else:
+            final_str = kv_translation(d[key], line + key_str, final_str)
+    return final_str
+
+
+def recursive_url_encode(data, parent_key=None):
+    items = []
+    for key, value in data.items():
+        if parent_key is not None:
+            new_key = f"{parent_key}[{key}]"
+        else:
+            new_key = key
+
+        if isinstance(value, dict):
+            items.extend(recursive_url_encode(value, new_key))
+        else:
+            items.append((new_key, value))
+
+    return items
+
+
+def is_customer_already_created(customer_name: str, stripe_api: StripeAPI):
+    customer = stripe_api.search_customer(customer_name)
+    return len(customer) > 0
+
+
+def create_customer_and_bank_account(customer: Customer, stripe_api: StripeAPI):
+    if customer.id is not None:
+        raise ValueError(f"Cannot create customer {customer} because it already has an ID")
+    if customer.bank_account and customer.bank_account.id is not None:
+        raise ValueError(f"Cannot create bank account for {customer} because it already has an ID")
+    customer_output = stripe_api.create_customer(customer)
+    if "id" in customer_output:
+        customer.id = customer_output["id"]
```

Comment:
nit: It feels dangerous to have partially instantiated objects. It means we need to have all those `if not customer.id: raise ValueError` I think it is because we blended the layer between the presentation layer and the domain. That being said, I don't expect the tooling to grow with a huge domain so it's probably not an issue